api/gateway/validation.go

```diff
@@ -23,14 +23,30 @@ func (m *Status) Validate() error {
 	return nil
 }
 
+// Location metadata errors
+var (
+	ErrInvalidLatitude  = errors.NewErrInvalidArgument("LocationMetadata", "invalid latitude")
+	ErrInvalidLongitude = errors.NewErrInvalidArgument("LocationMetadata", "invalid longitude")
+	ErrLocationZero     = errors.NewErrInvalidArgument("LocationMetadata", "is zero, so should be nil")
+)
+
 // Validate implements the api.Validator interface
-func (m *GPSMetadata) Validate() error {
-	if m == nil || m.IsZero() {
-		return errors.NewErrInvalidArgument("GPSMetadata", "can not be empty")
+func (m *LocationMetadata) Validate() error {
+	if m.IsZero() {
+		return ErrLocationZero
+	}
+	if m.Latitude > 90 || m.Latitude < -90 {
```

Comment:
I would even consider writing it like this:
`if m.Latitude > (90 - delta) || m.Latitude < (-90 + delta) {`

Because it is really very unlikely for a gateway to be located exactly on the pole, and is most likely a miss-configuration (that I've see a few times).